File: app/src/main/java/com/sabo/sabostorev2/API/APIRequestData.java
package com.sabo.sabostorev2.API;

import com.sabo.sabostorev2.Model.ResponseModel;

import okhttp3.MultipartBody;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;

public interface APIRequestData {

    /**
     * Sign Up
     */
    @FormUrlEncoded
    @POST("signUp.php")
    Call<ResponseModel> signUp(@Field("email") String email,
                               @Field("username") String username,
                               @Field("password") String password);


    /**
     * Sign In
     */
    @FormUrlEncoded
    @POST("signIn.php")
    Call<ResponseModel> signIn(@Field("email") String email,
                               @Field("username") String username,
                               @Field("password") String password);

    /**
     * Sign Out
     */
    @FormUrlEncoded
    @POST("signOut.php")
    Call<ResponseModel> signOut(@Field("uid") String uid);



    /**
     * Forgot Password
     */
    @FormUrlEncoded
    @POST("checkEmailExist.php")
    Call<ResponseModel> checkEmailExist(@Field("email") String email);

    @FormUrlEncoded
    @POST("forgotPassword.php")
    Call<ResponseModel> forgotPassword(@Field("email") String email,
                                       @Field("password") String password);

    /**
     * Get User Information
     */
    @FormUrlEncoded
    @POST("getUser.php")
    Call<ResponseModel> getUser(@Field("uid") String uid);


    /**
     * Update User Profile
     * - Email
     * - Username
     * - New Password
     */
    @FormUrlEncoded
    @POST("updateUserProfile.php")
    Call<ResponseModel> updateUserProfile(@Field("uid") String uid,
                                          @Field("oldPassword") String oldPassword,
                                          @Field("email") String email,
                                          @Field("username") String username,
                                          @Field("newPassword") String newPassword);

    /**
     * Update User Gender
     * - Gender
     */
    @FormUrlEncoded
    @POST("updateUserGender.php")
    Call<ResponseModel> updateUserGender(@Field("uid") String uid,
                                         @Field("gender") int gender);


    /**
     * Update User Image
     */
    @Multipart
    @POST("updateUserImage.php")
    Call<ResponseModel> updateUserImage(@Part MultipartBody.Part uid,
                                        @Part MultipartBody.Part file,
                                        @Part MultipartBody.Part oldFileName);

    /**
     * Remove User Image
     */
    @FormUrlEncoded
    @POST("removeUserImage.php")
    Call<ResponseModel> removeUserImage(@Field("uid") String uid,
                                        @Field("image") String image);

    /**
     * Get OTP
     * Update User Phone Number
     */
    @FormUrlEncoded
    @POST("getOTP.php")
    Call<ResponseModel> getOTP(@Field("uid") String uid,
                               @Field("phone") String phone,
                               @Field("code") int code);


    /**
     * Remove User Account
     */
    @FormUrlEncoded
    @POST("removeAccount.php")
    Call<ResponseModel> removeAccount(@Field("uid") String uid);


    /**
     * Delivery Cost
     */
    @FormUrlEncoded
    @POST("getDelivery.php")
    Call<ResponseModel> getDeliveryCost(@Field("code") String code);


    /**
     * Slider
     */
    @GET("getSlider.php")
    Call<ResponseModel> getSlider();


    /**
     * ItemStore
     */
    @GET("getItemStore.php")
    Call<ResponseModel> getItemStore();


    /**
     * Items
     */
    @FormUrlEncoded
    @POST("getItems.php")
    Call<ResponseModel> getItems(@Field("itemId") String itemId);


    /**
     * Item Details
     */
    @FormUrlEncoded
    @POST("getItemDetails.php")
    Call<ResponseModel> getItemsDetails(@Field("id") String id);


}

File: app/src/main/java/com/sabo/sabostorev2/API/APIRequestData.kt
package com.sabo.sabostorev2.API


import com.sabo.sabostorev2.Model.ExchangeRates.Currency
import com.sabo.sabostorev2.Model.ResponseModel
import okhttp3.MultipartBody
import retrofit2.Call
import retrofit2.http.*


interface APIRequestData {

    /**
     * ExchangeRatesAPI
     */

    @GET("latest?base=USD")
    fun getExchangeRatesAPI(): Call<Currency>


    /**
     * Sign Up
     */
    @FormUrlEncoded
    @POST("signUp.php")
    fun signUp(@Field("email") email: String?,
               @Field("username") username: String?,
               @Field("password") password: String?): Call<ResponseModel>


    /**
     * Sign In
     */
    @FormUrlEncoded
    @POST("signIn.php")
    fun signIn(@Field("email") email: String?,
               @Field("username") username: String?,
               @Field("password") password: String?): Call<ResponseModel>

    /**
     * Sign In with PIN
     */
    @FormUrlEncoded
    @POST("signInWithPIN.php")
    fun signInWithPIN(@Field("uid") uid: String?,
                      @Field("PIN") PIN: String?): Call<ResponseModel>

    /**
     * Sign Out
     */
    @FormUrlEncoded
    @POST("signOut.php")
    fun signOut(@Field("uid") uid: String?): Call<ResponseModel>


    /**
     * Forgot Password
     */
    @FormUrlEncoded
    @POST("checkEmailExist.php")
    fun checkEmailExist(@Field("email") email: String?): Call<ResponseModel>

    @FormUrlEncoded
    @POST("forgotPassword.php")
    fun forgotPassword(@Field("email") email: String?,
                       @Field("password") password: String?): Call<ResponseModel>

    /**
     * Get User Information
     */
    @FormUrlEncoded
    @POST("getUser.php")
    fun getUser(@Field("uid") uid: String?): Call<ResponseModel>


    /**
     * Update User EmailUsername
     * - Email
     * - Username
     */
    @FormUrlEncoded
    @POST("updateUserEmailUsername.php")
    fun updateUserEmailUsername(@Field("uid") uid: String?,
                                @Field("password") password: String?,
                                @Field("email") email: String?,
                                @Field("username") username: String?): Call<ResponseModel>

    /**
     * Update User Nickname
     */
    @FormUrlEncoded
    @POST("updateUserNickname.php")
    fun updateUserNickname(@Field("uid") uid: String?,
                           @Field("password") password: String?,
                           @Field("nickname") nickname: String?): Call<ResponseModel>


    /**
     * Update User Gender
     * - Gender
     */
    @FormUrlEncoded
    @POST("updateUserGender.php")
    fun updateUserGender(@Field("uid") uid: String?,
                         @Field("gender") gender: Int): Call<ResponseModel>


    /**
     * Update User Image
     */
    @Multipart
    @POST("updateUserImage.php")
    fun updateUserImage(@Part uid: MultipartBody.Part?,
                        @Part file: MultipartBody.Part?,
                        @Part oldFileName: MultipartBody.Part?): Call<ResponseModel>

    /**
     * Remove User Image
     */
    @FormUrlEncoded
    @POST("removeUserImage.php")
    fun removeUserImage(@Field("uid") uid: String?,
                        @Field("image") image: String?): Call<ResponseModel>

    /**
     * Get OTP
     */
    @FormUrlEncoded
    @POST("getOTP.php")
    fun getOTP(@Field("uid") uid: String?,
               @Field("phone") phone: String?,
               @Field("code") code: Int): Call<ResponseModel>


    /**
     * Update User Phone Number
     */
    @FormUrlEncoded
    @POST("updateUserPhone.php")
    fun updateUserPhone(@Field("uid") uid: String?,
                        @Field("phone") phone: String?,
                        @Field("countryCode") countryCode: String?): Call<ResponseModel>

    /**
     * UpdateUserPIN
     */
    @FormUrlEncoded
    @POST("updateUserPIN.php")
    fun updateUserPIN(@Field("uid") uid: String?,
                      @Field("isPIN") isPIN: Int?,
                      @Field("PIN") PIN: String?): Call<ResponseModel>


    /**
     * Remove User Account
     */
    @FormUrlEncoded
    @POST("removeAccount.php")
    fun removeAccount(@Field("uid") uid: String?,
                      @Field("email") email: String?,
                      @Field("password") password: String?): Call<ResponseModel>


    /**
     * Reset Password
     */
    @FormUrlEncoded
    @POST("resetPassword.php")
    fun resetPassword(@Field("uid") uid: String?,
                      @Field("newPassword") newPassword: String?): Call<ResponseModel>


    /**
     * Delivery Cost
     */
    @FormUrlEncoded
    @POST("getDelivery.php")
    fun getDeliveryCost(@Field("code") code: String?): Call<ResponseModel>


    /**
     * Slider
     */
    @GET("getSlider.php")
    fun getSlider(): Call<ResponseModel>


    /**
     * ItemSearch
     */
    @GET("getItemSearch.php")
    fun getItemSearch(): Call<ResponseModel>


    /**
     * Item Most Popular
     */
    @GET("getMostPopular.php")
    fun getMostPopular(): Call<ResponseModel>


    /**
     * ItemStore
     */
    @GET("getItemStore.php")
    fun getItemStore(): Call<ResponseModel>


    /**
     * Items
     */
    @FormUrlEncoded
    @POST("getItems.php")
    fun getItems(@Field("itemId") itemId: String?): Call<ResponseModel>


//    /**
//     * Item Details
//     */
//    @FormUrlEncoded
//    @POST("getItemDetails.php")
//    fun getItemsDetails(@Field("id") id: String?): Call<ResponseModel>


    /**
     * Items ORDER
     */
    @FormUrlEncoded
    @POST("inOrders.php")
    fun order(@Field("orderID") orderID: String?,
              @Field("uid") uid: String?,
              @Field("totalPrice") totalPrice: Double?): Call<ResponseModel>


    @FormUrlEncoded
    @POST("inOrderDetails.php")
    fun orderDetail(@Field("orderID") orderID: String?,
                    @Field("itemName") itemName: String?,
                    @Field("itemQuantity") itemQuantity: Int?,
                    @Field("itemPrice") price: Double?,
                    @Field("itemImage") itemImage: String?): Call<ResponseModel>


    @FormUrlEncoded
    @POST("getOrders.php")
    fun getOrders(@Field("uid") uid: String?): Call<ResponseModel>


    @FormUrlEncoded
    @POST("getOrdersByStatus.php")
    fun getOrdersByStatus(@Field("uid") uid: String?,
                          @Field("orderStatus") orderStatus: Int?): Call<ResponseModel>


    @FormUrlEncoded
    @POST("getOrderDetails.php")
    fun getOrderDetails(@Field("orderID") orderID: String?): Call<ResponseModel>
}